activesupport/lib/active_support/message_verifier.rb

```diff
@@ -74,8 +77,8 @@ def valid_message?(signed_message)
     def verified(signed_message)
       if valid_message?(signed_message)
         begin
-          data = signed_message.split("--")[0]
-          @serializer.load(decode(data))
+          data = signed_message.split(".")[1]
+          @serializer.load(decode(data)[0]).symbolize_keys
```

Comment:
Well, you shouldn’t call `symbolize_keys` after you’ve made the hash have indifferent access :smile:

> Den 25/06/2015 kl. 14.15 skrev Siddharth Bhatore <notifications@github.com>:
> 
> In activesupport/lib/active_support/message_verifier.rb <https://github.com/sbhatore/rails/pull/1#discussion_r33247915>:
> 
> > @@ -74,8 +77,8 @@ def valid_message?(signed_message)
> >      def verified(signed_message)
> >        if valid_message?(signed_message)
> >          begin
> > -          data = signed_message.split("--")[0]
> > -          @serializer.load(decode(data))
> > +          data = signed_message.split(".")[1]
> > +          @serializer.load(decode(data)[0]).symbolize_keys
> I tried to use HashWithIndifferentAccess like this here
> 
> HashWithIndifferentAccess.new(@serializer.load(decode(data)[0])).symbolize_keys
> but I don't think its returning the same to the verify function below in MessageVerifier. The tests are failing when I use claims['pld'] instead of claims[:pld] in verify! function of Claims .
> https://github.com/sbhatore/rails/pull/1/files#diff-d4289d86f57606dc70480330144996d3R26 <https://github.com/sbhatore/rails/pull/1/files#diff-d4289d86f57606dc70480330144996d3R26>.
> The claims variable comes from here
> https://github.com/sbhatore/rails/pull/1/files#diff-0da514c8bf2ba2f670bddf6514a52fa7R103 <https://github.com/sbhatore/rails/pull/1/files#diff-0da514c8bf2ba2f670bddf6514a52fa7R103>
> I will look more into this afterwards probably.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/sbhatore/rails/pull/1/files#r33247915>.
> 

--
Kasper